### PR TITLE
Fix/seeds

### PR DESCRIPTION
**Pull request by Muennighoff**
Fixes https://github.com/embeddings-benchmark/mteb/issues/13

To be safe, it initializes seeds in both the evaluator & the task super class. In case ever only one or the other is used.

We should have done this before running >50% of the evaluation 😂 I don't think think it's worth rerunning.

### CODE REVIEW

File: mteb/evaluation/evaluators/Evaluator.py [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
     """
     Base class for all evaluators
     Extend this class and implement __call__ for custom evaluators.
     """
+    def __init__(self, **kwargs):
+        self.seed = kwargs.get("seed", 42)
+
+        random.seed(self.seed)
+        np.random.seed(self.seed)
+        torch.manual_seed(self.seed)
```
Comment (NouamaneTazi) [inline mteb/evaluation/evaluators/Evaluator.py:17]: Should we add `if use_cuda: torch.cuda.manual_seed_all(seed_value)` like in [this example?](https://discuss.pytorch.org/t/what-is-manual-seed/5939/16)

File: mteb/abstasks/AbsTask.py [hunk 2]
```diff
@@ -1,7 +1,9 @@
 from abc import ABC, abstractmethod
+import random
 
 import datasets
-
+import numpy as np
+import torch
 
 class AbsTask(ABC):
     def __init__(self, **kwargs):
```
Comment (NouamaneTazi) [inline mteb/abstasks/AbsTask.py:9]:
```suggestion
    def __init__(self, seed=42, **kwargs):
```

File: mteb/abstasks/AbsTask.py [hunk 3]
```diff
@@ -11,6 +13,13 @@ def __init__(self, **kwargs):
         self.is_crosslingual = False
         self.save_suffix = kwargs.get("save_suffix", "")
 
+        self.seed = kwargs.get("seed", 42)
```
Comment (NouamaneTazi) [inline mteb/abstasks/AbsTask.py:16]:
```suggestion
        self.seed = seed
```

File: mteb/evaluation/evaluators/Evaluator.py [hunk 4]
```diff
@@ -1,11 +1,20 @@
...
 
+import numpy as np
+import torch
 
 class Evaluator(ABC):
     """
     Base class for all evaluators
     Extend this class and implement __call__ for custom evaluators.
     """
+    def __init__(self, **kwargs):
```
Comment (NouamaneTazi) [inline mteb/evaluation/evaluators/Evaluator.py:12]:
```suggestion
    def __init__(self, seed=42, **kwargs):
```

File: mteb/evaluation/evaluators/Evaluator.py [hunk 5]
```diff
@@ -1,11 +1,20 @@
...
+        self.seed = kwargs.get("seed", 42)
```
Comment (NouamaneTazi) [inline mteb/evaluation/evaluators/Evaluator.py:13]:
```suggestion
        self.seed = seed
```

File: mteb/abstasks/AbsTask.py [hunk 6]
```diff
@@ -11,6 +13,13 @@ def __init__(self, **kwargs):
...
+
+        random.seed(self.seed)
+        np.random.seed(self.seed)
+        torch.manual_seed(self.seed)
+
```
Comment (NouamaneTazi) [inline mteb/abstasks/AbsTask.py:21]: same here for `torch.cuda.manual_seed_all`

**Review by NouamaneTazi — CHANGES_REQUESTED** [review]
Haha, sorry I thought I had already fixed this issue. This explains why I never bothered to run all classification tasks. 😅
But hopefully this issue only concerns classification tasks (although we may want to re-run evaluation for all tasks anyway after the many modifications we've been doing)

**Review by NouamaneTazi — APPROVED** [review]
